modules/adagioAnalyticsAdapter.js

```diff
@@ -24,24 +42,287 @@ function canAccessTopWindow() {
   } catch (error) {
     return false;
   }
-}
+};
+
+function getCurrentWindow() {
+  return currentWindow;
+};
+
+let currentWindow;
+
+const adagioEnqueue = function adagioEnqueue(action, data) {
+  getCurrentWindow().ADAGIO.queue.push({ action, data, ts: Date.now() });
+};
+
+/**
+* END ADAGIO.JS CODE
+*/
+
+/**
+* UTILS FUNCTIONS
+*/
+
+const guard = {
+  adagio: (value) => isAdagio(value),
+  bidTracked: (auctionId, adUnitCode) => deepAccess(cache, `auctions.${auctionId}.${adUnitCode}`, false)
+};
+
+function removeDuplicates(arr, getKey) {
+  const seen = {};
+  return arr.filter(item => {
+    const key = getKey(item);
+    return seen.hasOwnProperty(key) ? false : (seen[key] = true);
+  });
+};
+
+function getAdapterNameForAlias(aliasName) {
+  return adapterManager.aliasRegistry[aliasName] || aliasName;
+};
+
+function isAdagio(value) {
+  return value.toLowerCase().includes('adagio') ||
+    getAdapterNameForAlias(value).toLowerCase().includes('adagio');
+};
+
+function getMediaTypeAlias(mediaType) {
+  const mediaTypesMap = {
+    banner: 'ban',
+    outstream: 'vidout',
+    instream: 'vidin',
+    adpod: 'vidadpod',
+    native: 'nat'
+  };
+  return mediaTypesMap[mediaType] || mediaType;
+};
+
+/**
+* sendRequest to Adagio. It filter null values and encode each query param.
+* @param {Object} qp
+*/
+function sendRequest(qp) {
+  // Removing null values
+  qp = Object.keys(qp).reduce((acc, key) => {
+    if (qp[key] !== null) {
+      acc[key] = qp[key];
+    }
+    return acc;
+  }, {});
+
+  const url = `${ENDPOINT}?${Object.keys(qp).map(key => `${key}=${enc(qp[key])}`).join('&')}`;
+  ajax(url, null, null, {method: 'GET'});
+};
+
+/**
+ * Send a new beacon to Adagio. It increment the version of the beacon.
+ * @param {string} auctionId
+ * @param {string} adUnitCode
+ */
+function newBeacon(auctionId, adUnitCode) {
```

Comment:
I would name this `sendBeacon`

Reply:
I renamed it `sendNewBeacon`, I prefer this because we not only sending it, but creating a new version of our beacon.